### PR TITLE
Gabbytang/search events

### PR DESCRIPTION
**Pull request by gabbytang**
## Trello ticket link

[As a user, I can search for events so that I can find relevant topics.](https://trello.com/c/6tWLDA45)

## Implementation description
* Implemented search feature to the events page
* Users can now type anything in the search bar and get a resulting list events that have either names, description, or creators that match the search term. 
* Created Search and SubmitButton components to be imported into other components. 
* *UPDATED:* Added clear search function which resets the search bar and search results 
   * Moved search button to be aligned with the text bar 
   * Integrated same changes to the Network page
   * Removed SubmitButton file and combined it with the Search file 

## Steps to test
1. Ensure the search field is open to any text value 
2. Clicking the search button should render the list of all events that match the input search term 
3. If nothing is entered in the search bar, the whole list of events in the db should be rendered
4. The search function should match an event by either name, description, or creator name
5. Make sure the "clear search" button works 

## What should reviewers focus on?
* The search function is accurate
* If there are no matches for a search, the page should be empty

### CODE REVIEW

**Review by Eshah7 — CHANGES_REQUESTED** [review]
Looks good!

Just a couple of questions: 

Why does Eng Student Working show up when I type in Te? 
Screenshot 2023-03-17 at 11 26 40 AM

Also, I was wondering if we should say "Search for events by name or user name" so that users know that both are possible separately? 

Thanks!

**Review by Eshah7 — APPROVED** [review]
LGTM!

File: src/components/common/Search.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import TextField from '@material-ui/core/TextField';
```
Comment (safchow) [inline src/components/common/Search.js:1]: nice! making these reusable is good practice 👍

**Review by safchow — COMMENTED** [review]
everything looks good!

In terms of the search functionality, what are your thoughts on adding a "Reset" or "Reset Search" option? The two ways to do this currently are to refresh the page or to search again but with no filter, which may not be intuitive for the user.

File: src/components/events/Events.js [hunk 2]
```diff
@@ -193,7 +202,36 @@ const Events = () => {
         </Typography>
       </Box>
 
-      <Box sx={{ position: 'absolute', top: 150, left: '50%', transform: 'translateX(-50%)' }}>
+      <Box sx={{ width:'600px', position: 'absolute', top: 150, left: '50%', transform: 'translateX(-50%)', marginBottom: '20px' }}>
+
+        <Search
```
Comment (safchow) [inline src/components/events/Events.js:206]: We should consider adding these searches to the "My ..." (ex "My Events") as well for consistency

**Review by safchow — CHANGES_REQUESTED** [review]

**Review by safchow — APPROVED** [review]
lgtm! Eventually we're going to need to style the search component (updating font, changing button colour, etc.) so its great that you made it a reusable component 👍 

As for when to style it, you can style it with this PR or merge this in and we can revisit styling later. Up to you. If you want to style it now, I can give you a walkthrough of CSS and Chakra UI whenever you're ready

**Pull request merged**